client/mysqldump.c

```diff
@@ -1200,7 +1201,8 @@ static int get_options(int *argc, char ***argv)
   if (opt_master_data)
   {
     opt_lock_all_tables= !opt_single_transaction;
-    opt_slave_data= 0;
+    if (opt_master_data != MYSQL_OPT_MASTER_DATA_COMMENTED_SQL)
```

Comment:
What do you expect to get in case `dump-slave=2` and `master-data= 1` and why?

Reply:
]]

> What do you expect to get in case `dump-slave=2` and `master-data= 1` and why, it is not clear for me?

A commented `dump-slave` and a non-commented `master-data`.


> There is `rpl_mysqldump_slave` test that is missing `dump-slave=2, master-data=2` case and this could be the point that we could compare results for this patch.

There now as 4.

> I think we don't need this line if (opt_master_data != MYSQL_OPT_MASTER_DATA_COMMENTED_SQL) since , opt_slave_data is checked first


At this stage in `get_options`, we haven't done anything the output. This is saying that if --dump-slave=1 --master-data=1, then dump-slave is effectively disabled. We cannot have both forms uncommented.